app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -392,6 +392,7 @@ class NavigationOverlayFragment : Fragment(), View.OnClickListener {
             NavigationEvent.TURBO -> toolbarViewModel.turboButtonClicked()
             NavigationEvent.DESKTOP_MODE -> toolbarViewModel.desktopModeButtonClicked()
             NavigationEvent.EXIT_FIREFOX -> {
+                toolbarViewModel.exitFirefoxButtonClicked()
                 (activity as MainActivity).moveTaskToBack(true)
             }
```

Comment:
This probably sounds ridiculous given how simple this case is, but simple cases can quickly build up into spaghetti code.

It's important to keep our data flows consistent, and be very strict about only allowing objects to touch things that are their responsibilities.  In this code, the `NavigationOverlayFragment` knows that clicks to the 'exit Firefox' button mean we need to exit the app.  But what if we want to reuse this exit code elsewhere, based on some other input?  Or if we're looking through the `ViewModel` trying to get a quick look at everything the overlay screen does, not knowing there's a _little_ bit of extra logic tucked within the `View`?  The `ViewModel` is handling the telemetry here, which is good, but it should handle all logic for the `View`, even something as simple as this.  

Again, a little silly for something this small, but consistency matters.  I think this code should call the VM and do nothing else.  Then the VM can send an event down the `events` LiveData, which can handle actually closing the app.